src/tests/multihost/alltests/test_krb_access_provider.py

```diff
@@ -185,22 +182,19 @@ def test_0003_k5login_user3_with_user4(multihost):
         for command in echo_cmds:
             cmd = multihost.client[0].run_command(command, raiseonerr=False)
 
-        user = 'foo3'
-        ssh_output = client.auth_from_client(user, 'Secret123')
-
+        ssh = check_login_client_bool(multihost, "foo3", "Secret123")
         client_hostname = multihost.client[0].sys_hostname
-        ssh = pexpect_ssh(client_hostname, 'foo4', 'Secret123', debug=False)
-        ssh.login(login_timeout=10, sync_multiplier=1, auto_prompt_reset=False)
-        ssh.command(f"ssh -o StrictHostKeyChecking=no -o PasswordAuthentication=no foo3@{client_hostname}\
-                     'id > /tmp/accessProvider_id_krb5_003.out 2>&1' ")
-        ssh.logout()
+        run_command_client(multihost, "foo4", "Secret123",
```

Comment:
The run_command does not have any error handling.
If the ssh inside fails it crashes on exception and cleanup will not be done.

It might be a good idea to put this to the try: except: finally: with the truncate in the finally clause (see other comment).

Even better solution might be extending the run_command with raiseonerr parameter that will allow catching/handling the exception inside and add a bool return if the operation succeeded.